Documentation/server-reflection-tutorial.md

```diff
@@ -0,0 +1,141 @@
+# gRPC Server Reflection Tutorial
+
+gRPC Server Reflection provides information about publicly-accessible gRPC
+services on a server, and assists clients at runtime to construct RPC
+requests and responses without precompiled service information. It is used by
+gRPC CLI, which can be used to introspect server protos and send/receive test
+RPCs.
+
+## Enable Server Reflection
+
+### Enable server reflection in gRPC-go servers
+
+gRPC-go Server Reflection is implemented in package [reflection](https://github.com/grpc/grpc-go/tree/master/reflection). To enable server reflection, you need to import this package and register reflection service on your gRPC server.
+
```

Comment:
Instead of the hypothetical example below, do you want to state that this example walks through how server reflection can be enabled on the hello world example?

We can then use real code such as 
``pb.RegisterGreeterServer(s, &server{})`` instead of the functions below.